### PR TITLE
app-emulation/wine-staging: Add wine-staging-6.0_rc4

### PR DESCRIPTION
**Pull request by FireBurn**
This allows us to test the rc's, it already includes the src_pretend change which allows me to install a bin package with +mingw on a machine without mingw setup (I've confirmed it works)

This can probably be used with wine-vanilla too

Signed-off-by: Mike Lothian <mike@fireburn.co.uk>

### CODE REVIEW

File: app-emulation/wine-staging/wine-staging-6.0_rc4.ebuild [hunk 1]
```diff
@@ -0,0 +1,663 @@
...
+	if [[ ! -z ${EGIT_COMMIT} ]]; then
+		eerror "Commits must now be specified using the environment variables:"
+		eerror "  EGIT_OVERRIDE_COMMIT_WINE"
+		eerror "  EGIT_OVERRIDE_COMMIT_WINE_STAGING_WINE_STAGING"
+		eerror
+		return 1
+	fi
+}
+
+src_pretend() {
```
Comment (sarnex) [inline app-emulation/wine-staging/wine-staging-6.0_rc4.ebuild:291]: Please rename to pkg_pretend so it gets hit

File: app-emulation/wine-staging/wine-staging-6.0_rc4.ebuild [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+	wine_build_environment_check || die
+
+	# Verify OSS support
+	if use oss && ! use kernel_FreeBSD; then
+		if ! has_version ">=media-sound/oss-4"; then
```
Comment (sarnex) [inline app-emulation/wine-staging/wine-staging-6.0_rc4.ebuild:296]: Are there any other changes to the ebuild besides pkg_pretend->src_pretend?

**Review by sarnex — CHANGES_REQUESTED** [review]

File: app-emulation/wine-staging/wine-staging-6.0_rc4.ebuild (hunk 2 above)
Comment (FireBurn) [reply app-emulation/wine-staging/wine-staging-6.0_rc4.ebuild:296]:
```
--- /usr/portage/app-emulation/wine-staging/wine-staging-5.22-r2.ebuild 2020-12-05 09:24:53.880476216 +0000
+++ wine-staging-6.0_rc4.ebuild 2020-12-31 22:21:52.447298487 +0000
@@ -7,11 +7,11 @@
 PLOCALE_BACKUP="en"
 
 inherit autotools eapi7-ver estack eutils flag-o-matic gnome2-utils l10n multilib multilib-minimal pax-utils toolchain-funcs virtualx xdg-utils
-
 MY_PN="${PN%%-*}"
-MY_P="${MY_PN}-${PV}"
+MY_PV="${PV/_/-}"
+MY_P="${MY_PN}-${MY_PV}"
 
-if [[ ${PV} == "9999" ]] ; then
+if [[ ${MY_PV} == "9999" ]] ; then
        EGIT_REPO_URI="https://source.winehq.org/git/wine.git"
        EGIT_BRANCH="master"
        inherit git-r3
@@ -19,12 +19,16 @@
        #KEYWORDS=""
 else
        MAJOR_V=$(ver_cut 1)
-       SRC_URI="https://dl.winehq.org/wine/source/${MAJOR_V}.x/${MY_P}.tar.xz"
+       MINOR_V=$(ver_cut 2)
+    if [[ ${MINOR_V} != "0" ]] ; then
+        MINOR_V="x"
+    fi
+       SRC_URI="https://dl.winehq.org/wine/source/${MAJOR_V}.${MINOR_V}/${MY_P}.tar.xz"
        KEYWORDS="-* ~amd64 ~x86"
 fi
 S="${WORKDIR}/${MY_P}"
 
-STAGING_P="wine-staging-${PV}"
+STAGING_P="wine-staging-${MY_PV}"
 STAGING_DIR="${WORKDIR}/${STAGING_P}"
 GWP_V="20200523"
 PATCHDIR="${WORKDIR}/gentoo-wine-patches"
@@ -35,15 +39,15 @@
        https://dev.gentoo.org/~sarnex/distfiles/wine/gentoo-wine-patches-${GWP_V}.tar.xz
 "
 
-if [[ ${PV} == "9999" ]] ; then
+if [[ ${MY_PV} == "9999" ]] ; then
        STAGING_EGIT_REPO_URI="https://github.com/wine-staging/wine-staging.git"
 else
        SRC_URI="${SRC_URI}
-       staging? ( https://github.com/wine-staging/wine-staging/archive/v${PV}.tar.gz -> ${STAGING_P}.tar.gz )"
+       staging? ( https://github.com/wine-staging/wine-staging/archive/v${MY_PV}.tar.gz -> ${STAGING_P}.tar.gz )"
 fi
 
 LICENSE="LGPL-2.1"
-SLOT="${PV}"
+SLOT="${MY_PV}"
 IUSE="+abi_x86_32 +abi_x86_64 +alsa capi cups custom-cflags dos elibc_glibc +faudio +fontconfig +gcrypt +gecko gphoto2 gsm gssapi gstreamer +jpeg kerberos kernel_FreeBSD +lcms ldap mingw +mono mp3 netapi nls odbc openal opencl +opengl osmesa oss +perl pcap pipelight +png prelink pulseaudio +realtime +run-exes samba scanner sdl selinux +ssl staging test themes +threads +truetype udev +udisks +unwind v4l vaapi vkd3d vulkan +X +xcomposite xinerama +xml"
 REQUIRED_USE="|| ( abi_x86_32 abi_x86_64 )
        X? ( truetype )
@@ -171,7 +175,7 @@
 PATCHES_BIN=()
 
 # https://bugs.gentoo.org/show_bug.cgi?id=635222
-if [[ ${#PATCHES_BIN[@]} -ge 1 ]] || [[ ${PV} == 9999 ]]; then
+if [[ ${#PATCHES_BIN[@]} -ge 1 ]] || [[ ${MY_PV} == 9999 ]]; then
        DEPEND+=" dev-util/patchbin"
 fi
 
@@ -284,7 +288,7 @@
        fi
 }
 
-pkg_pretend() {
+src_pretend() {
        wine_build_environment_check || die
 
        # Verify OSS support
@@ -330,7 +334,7 @@
        wine_build_environment_check || die
        wine_env_vcs_vars || die
 
-       WINE_VARIANT="${PN#wine}-${PV}"
+       WINE_VARIANT="${PN#wine}-${MY_PV}"
        WINE_VARIANT="${WINE_VARIANT#-}"
 
        MY_PREFIX="${EPREFIX}/usr/lib/wine-${WINE_VARIANT}"
@@ -344,7 +348,7 @@
 }
 
 src_unpack() {
-       if [[ ${PV} == "9999" ]] ; then
+       if [[ ${MY_PV} == "9999" ]] ; then
                EGIT_CHECKOUT_DIR="${S}" git-r3_src_unpack
                if use staging; then
                        local CURRENT_WINE_COMMIT=${EGIT_VERSION}
@@ -614,12 +618,12 @@
 }
 
 pkg_postinst() {
-       eselect wine register ${P}
+       eselect wine register ${PN}-${MY_PV}
        if [[ ${PN} == "wine-vanilla" ]]; then
-               eselect wine register --vanilla ${P} || die
+               eselect wine register --vanilla ${PN}-${MY_PV} || die
        else
                if use staging; then
-                       eselect wine register --staging ${P} || die
+                       eselect wine register --staging ${PN}-${MY_PV} || die
                fi
        fi
 
@@ -642,12 +646,12 @@
 }
 
 pkg_prerm() {
-       eselect wine deregister ${P}
+       eselect wine deregister ${PN}-${MY_PV}
        if [[ ${PN} == "wine-vanilla" ]]; then
-               eselect wine deregister --vanilla ${P} || die
+               eselect wine deregister --vanilla ${PN}-${MY_PV} || die
        else
                if use staging; then
-                       eselect wine deregister --staging ${P} || die
+                       eselect wine deregister --staging  ${PN}-${MY_PV}|| die
                fi
        fi
```

Comment (FireBurn) [reply app-emulation/wine-staging/wine-staging-6.0_rc4.ebuild:296]: That's the diff between 5.22-r2 and my ebuild

**Comment by sarnex** [comment]
Thanks. We don't usually release RC releases, so can you keep this just locally for now? Not sure I want to push it

**Comment by FireBurn** [comment]
I was hoping to change that. The RC's should be more stable then the 5.x releases and hopefully with the added testing might see a lot of old bugs resolved with our help

**Comment by sarnex** [comment]
OK yeah thats fine, please update this when you solve the src_pretend issue

**Comment by Chiitoo** [comment]
The RCs are so short-lived so I don't really see a point in spending the effort into them.

In a way, they're just 9999 after all, as far as I can tell.  :]

**Comment by FireBurn** [comment]
It's now just a rename to use an rc, even if you don't add in the rc ebuilds, it might be a good idea to use this code so they could be used going forward

I'm using these locally anyway it's up to yourselves if you'd like to run with it or not

**Comment by sarnex** [comment]
Can you please squash the comments into one

**Comment by FireBurn** [comment]
Sorted

**Pull request closed without merging**